### PR TITLE
Feature/test with rspec

### PR DESCRIPTION
**Pull request by Hermela-Fikre**
- Used existed repositories for tic tac toe.
- Each method was tested with the describe block.
- Used one it block for each expected behaviour.
- Fixed linter errors.

### CODE REVIEW

File: spec/test.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+  describe '#add_to_table' do
+    arr = Array[1, 2, 3, 4, 5, 6, 7, 8, 9]
+    random = rand(1..9)
+    arr.delete_at(random - 1)
+    expected = arr.insert(random - 1, nil)
+    it 'should add the player choice into a board' do
+      expect(game.add_to_table(random)).to eq(expected)
+    end
+  end
```
Comment (maelfosso) [inline spec/test.rb]: Check the content of the `game.board` at that position

File: spec/test.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+  describe '#valid_position?' do
+    random = rand(1..9)
+    it 'checks if selected position is valid' do
+      expect(game.valid_position?(random)).to eql(false)
+    end
+    it 'returns false when a string is used as an argument' do
+      expect(game.valid_position?('')).to be_falsey
+    end
+  end
```
Comment (maelfosso) [inline spec/test.rb]: Check also what happens when a value is set at a position

File: spec/test.rb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+  describe '#current_player' do
+    it 'switch players' do
+      expect(game.current_player(0)).to be true
+    end
+    it 'switch players' do
+      expect(game.current_player(1)).to_not be true
+    end
+  end
```
Comment (maelfosso) [inline spec/test.rb]: There is also a case that takes us to `False`.

File: spec/test.rb [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    it 'checks validity of players\'s choice' do
+      expect(Game.valid_letter?('O')).to be true
+    end
+    it 'Checks validity of players\'s choice' do
+      expect(Game.valid_letter?('X')).to be true
+    end
+    it 'checks validity of players\'s choice' do
+      expect(Game.valid_letter?('W')).to_not be true
+    end
+  end
```
Comment (maelfosso) [inline spec/test.rb]: The method `valid_letter` doesn't always return `True`. What about the other case?

File: spec/test.rb [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+describe Board do
+  describe '#winner?' do
+    context 'When there is a winner' do
+      let(:board) { ['O', 'O', 'O', ' ', ' ', ' ', ' ', ' ', ' '] }
+      let(:letter) { 'O' }
+      it 'return true' do
+        expect(Board.winner?(board, letter)).to eq true
+      end
+    end
+  end
```
Comment (maelfosso) [inline spec/test.rb]: What about the `False` situation?

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello @Hermela-Fikre @MrSayyod 

Nice introduction to the milestone.
However, there are some changes to do

1. When using RSpec, the file containing test must finish by `_spec.rb`

2. When running the test I got this error
```rb
$ rspec spec/test.rb 

An error occurred while loading ./spec/test.rb.
Failure/Error: require '../lib/game.rb'

LoadError:
  cannot load such file -- ../lib/game.rb
# ./spec/test.rb:1:in `<top (required)>'
No examples found.


Finished in 0.00002 seconds (files took 0.13248 seconds to load)
0 examples, 0 failures, 1 error occurred outside of examples
```
3. Please, read the comments left below.

Kindly make the suggested changes and submit a new code review request
Thanks

Mael FOSSO

File: spec/test.rb (hunk 3 above)
Comment (MrSayyod) [reply spec/test.rb]: Fixed

File: spec/test.rb (hunk 5 above)
Comment (MrSayyod) [reply spec/test.rb]: fixed

File: spec/test.rb (hunk 4 above)
Comment (MrSayyod) [reply spec/test.rb]: fixed

**Review by rOluochKe — APPROVED** [review]
Hello @MrSayyod and @Hermela-Fikre 
### You have done really well in this project and for also implementing changes from the previous reviews :clap: The project looks good and congratulations :boom: :100: it has been approved.
### You can merge and submit a project completion form from the dashboard. Good luck with the next project.
- Happy coding!

**Pull request merged**